File: core/src/main/java/com/harera/hayat/core/repository/GlobalMessageRepository.java
package com.harera.hayat.core.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.harera.hayat.model.exception.GlobalMessage;

@Repository
public interface GlobalMessageRepository extends JpaRepository<GlobalMessage, Long> {
    Optional<GlobalMessage> findByLanguageAndCode(String language, String code);
}

File: src/main/java/com/harera/hayatserver/repository/GlobalMessageRepository.kt
package com.harera.hayatserver.repository;

import com.harera.hayatserver.model.GlobalMessage
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface GlobalMessageRepository : JpaRepository<GlobalMessage, Long> {

    fun findByLanguageAndCode(language: String?, code: String?): Optional<GlobalMessage>
}
